devito/propagator.py

```diff
@@ -105,6 +105,7 @@ def convert_equality_to_cgen(self, equality):
         try:
             return cgen.Assign(ccode(self.time_substitutions(equality.lhs).xreplace(self._var_map)), ccode(self.time_substitutions(equality.rhs).xreplace(self._var_map)))
         except:
+            raise
```

Comment:
Not sure the `raise` here is intentional. Either raise a `RuntimeError` with an informative message, or don't catch the error at all.